MM-41988 Updated links to legacy domain about.mattermost.com

cwarnermm opened the pull request:
#### Summary
Updated in-product links to legacy domain about.mattermost.com.

#### Related Pull Requests

List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- Has docs changes: https://github.com/mattermost/docs/pull/5409
- Has dev docs changes: https://github.com/mattermost/mattermost-developer-documentation/pull/1012 & https://github.com/mattermost/mattermost-developer-documentation/pull/1019
- Has handbook changes: https://github.com/mattermost/mattermost-handbook/pull/962
- Has server changes: https://github.com/mattermost/mattermost-server/pull/19552
- Has mobile changes: https://github.com/mattermost/mattermost-mobile/pull/5962
- Has enterprise changes: https://github.com/mattermost/enterprise/pull/1147
- Has desktop changes: https://github.com/mattermost/desktop/pull/2003
- Has customer-web-server changes: https://github.com/mattermost/customer-web-server/pull/677
- Has mattermost-cloud changes: https://github.com/mattermost/mattermost-cloud/pull/619

#### Release Note
```release-note
Updated in-product links to legacy domain about.mattermost.com
```

cwarnermm commented:
**Important QA Note**: Running ``npm run updatesnapshot -i`` commands with my current work machine locks up my computer completely, leaving it unusable (even when I run a single command at a time). 

Here is the list of ``TSX`` files I've updated in this PR that won't necessarily have snapshots updated in parallel:

- npm run updatesnapshot -i ./components/about_build_modal/about_build_modal.tsx 
- npm run updatesnapshot -i./c[omponents/about_build_modal/about_build_modal_cloud/about_build_modal_cloud.tsx](https://github.com/mattermost/mattermost-webapp/blob/b857e24169eff605b0bde4731425133091528bc0/components/about_build_modal/about_build_modal_cloud/about_build_modal_cloud.tsx) 
- npm run updatesnapshot -i./components/admin_console/admin_navbar_dropdown/admin_navbar_dropdown.tsx 
- npm run updatesnapshot -i./[components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx](https://github.com/mattermost/mattermost-webapp/blob/59040a8a6a082c33be441ee514d9a9450af323cb/components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx) 
- npm run updatesnapshot -i./[components/admin_console/plugin_management/plugin_management.tsx](https://github.com/mattermost/mattermost-webapp/blob/1433271fa7e66b93a741b49878d2c1222ddcba0d/components/admin_console/plugin_management/plugin_management.tsx) 
- npm run updatesnapshot -i./components/system_notice/notices.tsx 
- npm run updatesnapshot -i./[components/admin_console/permission_schemes_settings/permission_schemes_settings.tsx](https://github.com/mattermost/mattermost-webapp/blob/be0f1e7e04af0925f96a829682363b08d758971f/components/admin_console/permission_schemes_settings/permission_schemes_settings.tsx) 
- npm run updatesnapshot -i./components/user_settings/security/user_access_token_section/user_access_token_section.tsx 
- npm run updatesnapshot -i./components/integrations/installed_oauth_apps/installed_oauth_apps.tsx
- npm run updatesnapshot -i./[components/admin_console/manage_roles_modal/manage_roles_modal.tsx](https://github.com/mattermost/mattermost-webapp/blob/74cbcb7c3f0dff7e21ecb88776b8841dd1fa55e6/components/admin_console/manage_roles_modal/manage_roles_modal.tsx) 
- npm run updatesnapshot -i./components/admin_console/license_settings/modals/ee_license_modal.tsx 
- npm run updatesnapshot -i./[components/integrations/installed_commands/installed_commands.tsx](https://github.com/mattermost/mattermost-webapp/blob/74cbcb7c3f0dff7e21ecb88776b8841dd1fa55e6/components/integrations/installed_commands/installed_commands.tsx) 
- npm run updatesnapshot -i./components/admin_console/permission_schemes_settings/permission_system_scheme_settings/permission_system_scheme_settings.tsx 
- npm run updatesnapshot -i./components/start_trial_modal/index.tsx 
- npm run updatesnapshot -i./components/integrations/installed_incoming_webhooks/installed_incoming_webhooks.tsx 
- npm run updatesnapshot -i./components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings.tsx

saturninoabril commented:
/e2e-test

mattermod commented:
Triggering e2e testing with options:
```<nil>```

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/153802

saturninoabril reviewed (changes requested):
Thanks Carrie! Looks good to me, except for few clarifications.

Also, if I search "about.mattermost.com", I still get some results in couple of components (`admin_definition.jsx` and `push_settings.jsx`). Could you please check if it's something to update as well. For other changes required in E2E tests, I'll help you there.

cwarnermm commented on components/admin_console/plugin_management/plugin_management.tsx:
```diff
@@ -976,21 +976,21 @@ export default class PluginManagement extends AdminSettings<Props, State> {
                 <FormattedMarkdownMessage
                     id='admin.plugin.uploadDesc'
-                    defaultMessage='Upload a plugin for your Mattermost server. See [documentation](!https://about.mattermost.com/default-plugin-uploads) to learn more.'
+                    defaultMessage='Upload a plugin for your Mattermost server. See [documentation](!https://developers.mattermost.com/integrate/admin-guide/admin-plugins-beta/) to learn more.'
```
Good catch! "Beta" labels were recently removed from dev.mm.com for plugins, so we can remove the "beta" reference here as well.

cwarnermm commented on components/admin_console/push_settings.jsx:
```diff
@@ -126,28 +126,28 @@ export default class PushSettings extends AdminSettings {
                 <FormattedMarkdownMessage
                     id='admin.email.pushOffHelp'
-                    defaultMessage='Please see [documentation on push notifications](!https://about.mattermost.com/default-mobile-push-notifications/) to learn more about setup options.'
+                    defaultMessage='Please see [documentation on push notifications](!https://docs.mattermost.com/deploy/mobile-hpns.html) to learn more about setup options.'
```
I agree that the URL for this docs page isn't as discoverable as it could be; however, let's leave this URL as-is for now.

cwarnermm commented:
@saturninoabril Thanks! I believe I've updated the remaining legacy URLs. I've observed two issues while making these updates:
- we have many ``http`` links that should be ``https``. These updates are currently out of scope; however, I've reached out to R&D to see if there's a programmatic way to approach this kind of update.
- I've seen a number of links that are essentially broken/malformed due to translation changes being applied. Correcting these is also currently out of scope. I've asked R&D if there's a linter or similar available to help catch these types of issues.

saturninoabril commented:
Thanks Carrie and well noted! I'll wait for the server and enterprise PRs to get merge. Once merged, let's update this branch so the build will pick up the latest changes from other repo then run E2E. I'll help fix other test afterward.

hmhealey commented on components/about_build_modal/about_build_modal.tsx:
```diff
@@ -278,7 +278,7 @@ export default class AboutBuildModal extends React.PureComponent<Props, State> {
                             <FormattedMarkdownMessage
                                 id='about.notice'
-                                defaultMessage='Mattermost is made possible by the open source software used in our [server](!https://about.mattermost.com/platform-notice-txt/), [desktop](!https://about.mattermost.com/desktop-notice-txt/) and [mobile](!https://about.mattermost.com/mobile-notice-txt/) apps.'
+                                defaultMessage='Mattermost is made possible by the open source software used in our [server](!https://github.com/mattermost/mattermost-server/blob/master/NOTICE.txt), [desktop](!https://github.com/mattermost/desktop/blob/master/NOTICE.txt) and [mobile](!https://github.com/mattermost/mattermost-mobile/blob/master/NOTICE.txt) apps.'
```
Note that these links may break if we ever move around the code in those repos. I think the reason that the old version pointed to about.mattermost.com was that we had redirects set up so that older versions of the app  would still link to the correct place in case the repos ever changed

cwarnermm replied:
Good point, @hmhealey. There will be redirects in place via Marketing following the switchover; however, ideally we don't want to reply on those alone. I'm 2/5 for applying these URL updates as well. Given the number of instances of these strings alone in the product, we should align on an approach.

@emdecr - What is your recommendation for situations like where we've got in-product links pointing to legacy domains intentionally to avoid future rework? Where would you say we draw the line between updated domain links and keeping existing legacy links in place?

emdecr replied:
@cwarnermm Hmm good to know. I'm leaning towards still replacing them since we're phasing out this subdomain entirely. We have the ability to update redirects, which is what would have to happen with this old setup (ie. if `about.mm.com/platform-notice-text/` needed to be pointed at something else, we'd have to update the redirect anyways).

What might help is to map out which URLs are considered as these types of stand-in links, and make mm.com equivalents. 

Does that make sense/help at all? Am I understanding the issue, or am I way off base?

hmhealey replied:
That sounds like it would help for if the linked pages ever change in the future (like if the NOTICE.txt in the GitHub repo moves or AWS moves a docs page that we're linking to), but it doesn't fix the issue that older instances will still be pointing at `about.mattermost.com` URLs so those links will break. That will however be fixed over time as people update to newer versions of MM.

The bigger problem I see is for the configurable links stored in the config.json since even if we change the defaults, they'll never be updated on their own, so we could end up with someone's config still pointing to `about.mattermost.com` even if they do update to a newer version of MM. We could go and make it so that those links are rewritten by the server code on upgrade though.

cwarnermm replied:
@hmhealey - This is ultimately a conversation and decision to be made jointly between Marketing and Engineering that balances potential risk against future-proofing. One key thing is that while these domains are going away for good, redirects _will_ be in place, so we have some time to make an appropriate future-proofed decision that doesn't have to be now.

What we do need to decide now, as part of the domain consolidation effort, is whether we keep these URL updates as-is or revert them in this repo.
- if we leave them as-is, and Engineering refactors code in the future, those links may break. That feels like the cost of a refactor in general.
- if we choose to revert them, I agree with @emdecr that we'll need to account for this in the future somehow since these domains are being sunsetted permanently. 

With this in mind, how do you both want to proceed, @hmhealey and @emdecr?

emdecr replied:
Thanks for the all the context @hmhealey.

Just to be safe, I'm fine with making the call to **revert back to keeping these about.mm.com links in-tact until we have more eyes/ears on this topic**. 

This probably merits a larger conversation once the dust settles re: replacing these URLs, and a strategy moving forward. Might be worth it to make a product-specific subdomain vs referencing any type of user-facing site. Then we can use that subdomain as the redirect hub for all these stand-in links.

We're not going to ax the about.mm.com redirects anytime soon either. So, people will end up at destinations vs being served a 404.

hmhealey commented on components/admin_console/admin_definition.jsx:
```diff
@@ -1104,7 +1104,7 @@ const AdminDefinition = {
                         help_text: t('admin.image.amazonS3IdDescription'),
                         help_text_markdown: true,
-                        help_text_default: '(Optional) Only required if you do not want to authenticate to S3 using an [IAM role](!https://about.mattermost.com/default-iam-role). Enter the Access Key ID provided by your Amazon EC2 administrator.',
+                        help_text_default: '(Optional) Only required if you do not want to authenticate to S3 using an [IAM role](!https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html). Enter the Access Key ID provided by your Amazon EC2 administrator.',
```
The same thing goes for many of these ones that are out of our direct control. Those ones I'd be even more worried about since at least we know when we're reorganizing the repos that stuff might break

cwarnermm replied:
@emdecr - This is a similar case to consider.

hmhealey commented on e2e/cypress/fixtures/markdown/markdown_inline_images_6.md:
```diff
@@ -1,3 +1,3 @@
 ### In-line Images
 
-![test image](https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)
+![test image](http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)
```
@saturninoabril Changing these URLs will likely break a lot of older tests around message attachments and Markdown images since the old URL used for them won't exist any more. I don't know where else we might need to change these

@cwarnermm Also, a bunch of these are changing https URLs to http which isn't ideal. That seems to be happening most in these E2E tests, but we should probably switch those back to using https

cwarnermm replied:
@emdecr - The new destination URLs for these are ``http`` and that may be a typo. Can you please confirm? In the event that it's intentional, please note @hmhealey's feedback, and let me know how best to proceed.

cwarnermm replied:
Confirmed with @emdecr that image assets should be ``https``. I'll ensure they're updated in this PR.

hmhealey commented on e2e/cypress/integration/files_and_attachments/image_link_preview_spec.js:
```diff
@@ -38,7 +38,7 @@ describe('Image Link Preview', () => {
 
     it('MM-T331 Image link preview - Collapse and expand', () => {
-        const link = 'https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png';
+        const link = 'http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png';
```
```suggestion
        const link = 'https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png';
```

cwarnermm replied:
```suggestion
        const link = 'https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png';
```

hmhealey commented on e2e/cypress/integration/integrations/incoming_webhook/disallow_username_profile_override_spec.js:
```diff
@@ -45,7 +45,7 @@ describe('Incoming webhook', () => {
 
     it('MM-T622 Disallow override of username and profile picture', () => {
-        const iconUrl = 'https://www.mattermost.org/wp-content/uploads/2016/04/icon.png';
+        const iconUrl = 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png';
```
```suggestion
        const iconUrl = 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png';
```

cwarnermm replied:
```suggestion
        const iconUrl = 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png';
```

hmhealey commented on e2e/cypress/integration/markdown/markdown_image_spec.js:
```diff
@@ -122,7 +122,7 @@ describe('Markdown', () => {
     it('channel header is markdown image', () => {
         // # Update channel header
-        cy.updateChannelHeader('![MM Logo](https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)').wait(TIMEOUTS.TWO_SEC);
+        cy.updateChannelHeader('![MM Logo](http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)').wait(TIMEOUTS.TWO_SEC);
```
```suggestion
        cy.updateChannelHeader('![MM Logo](https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)').wait(TIMEOUTS.TWO_SEC);
```

cwarnermm replied:
```suggestion
        cy.updateChannelHeader('![MM Logo](https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)').wait(TIMEOUTS.TWO_SEC);
```

hmhealey commented on e2e/cypress/support/api/on_prem_default_config.json:
```diff
@@ -231,9 +231,9 @@
-        "AboutLink": "https://about.mattermost.com/default-about/",
-        "HelpLink": "https://about.mattermost.com/default-help/",
-        "ReportAProblemLink": "https://about.mattermost.com/default-report-a-problem/",
+        "AboutLink": "https://docs.mattermost.com/about/product.html",
```
This'll be something we'd also need to change on the server, but changing these will actually break those links for all existing Mattermost instances that haven't customized them. The old about.mattermost.com URLs are saved in those servers' config files, so we need those redirects to still exist.

Even if we could change them for existing installs, I think it would be best to leave them as redirects anyway since we could end up changing the URLs for some of them. The download links for the mobile apps, in particular, will likely change when Mobile v2 comes out

cwarnermm replied:
@emdecr - Another good point to take into consideration.

emdecr replied:
Similar to my [previous comment](https://github.com/mattermost/mattermost-webapp/pull/9785#discussion_r814310975). Let's revert back to the old links.

hmhealey commented on i18n/bg.json:
```diff
@@ -13,7 +13,7 @@
   "about.hashwebapp": "Webapp билд хеш:",
   "about.licensed": "Лицензиран на:",
-  "about.notice": "Използването на Mattermost е възможно благодарение на софтуер с отворен код използван в нашите [сървърно](!https://about.mattermost.com/platform-notice-txt/), [настолно](!https://about.mattermost.com/desktop-notice-txt/) и [мобилно](!https://about.mattermost.com/mobile-notice-txt/) приложения.",
+  "about.notice": "Използването на Mattermost е възможно благодарение на софтуер с отворен код използван в нашите [сървърно](!https://github.com/mattermost/mattermost-server/blob/master/NOTICE.txt), [настолно](!https://github.com/mattermost/desktop/blob/master/NOTICE.txt) и [мобилно](!https://github.com/mattermost/mattermost-mobile/blob/master/NOTICE.txt) приложения.",
```
Note that this might cause issues if these conflict with the strings that are automatically submitted from the translation server. I'm not sure how we've dealt with stuff like this in the past since we've never done this sort of mass update before

hmhealey replied:
I think it'd have a similar effect on the translators since they'd still have to update the strings in this release, but afterwards it would mean we could change the URLs freely. I do think it would be a good idea though since it removes any risk of a URL getting changed accidentally by a translator.

Depending on how far we want to go with that, there's a way in the i18n library that we use where we could make the translated string look like
```
"some.string": "this is <link>a link to some web site</link>"
```
and then the code would be entirely responsible for what the link is and what it looks like as opposed to now where the translator has to deal with a Markdown directly

calebroseland replied:
> `"some.string": "this is <link>a link to some web site</link>"`
+1

cwarnermm replied:
Awesome! Thanks, @hmhealey! I'm capturing this as a future enhancement. 

In terms of the domain switchover we're facing in the short term, what would you recommend as next steps? Should I:
- leave my string updates as-is in this repo (and notify the translation community that a LOT of strings are going to look like they changed but have likely only changed in terms of the link URL)
- revert my string updates in all language files except ``en.json`` (and if we go this route, will those URL changes get picked up via a round-trip through Weblate, perhaps?)

cwarnermm commented on .storybook/theme.js:
```diff
@@ -35,6 +35,6 @@ export default create({
-    brandUrl: 'https://mattermost.org',
-    brandImage: 'https://www.mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png',
+    brandUrl: 'https://mattermost.com',
+    brandImage: 'http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png',
```
```suggestion
    brandImage: 'https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png',
```

cwarnermm commented on e2e/cypress/fixtures/markdown/markdown_inline_images_6.md:
```diff
@@ -1,3 +1,3 @@
 ### In-line Images
 
-![test image](https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)
+![test image](http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)
```
```suggestion
![test image](https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)
```

cwarnermm commented on e2e/cypress/integration/files_and_attachments/image_link_preview_new_window_spec.js:
```diff
@@ -30,7 +30,7 @@ describe('Image Link Preview', () => {
 
     it('MM-T329 Image link preview', () => {
-        const link = 'https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png';
+        const link = 'http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png';
```
```suggestion
        const link = 'https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png';
```

cwarnermm commented on e2e/cypress/integration/integrations/custom_slash_commands/custom_slash_commands_spec.js:
```diff
@@ -156,7 +156,7 @@ describe('Slash commands', () => {
         // # click on last added command's(first child) edit action
         cy.get('.backstage-list').find('.backstage-list__item').first().findByText('Edit').click();
-        const iconURL = 'http://www.mattermost.org/wp-content/uploads/2016/04/icon_WS.png';
+        const iconURL = 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png';
```
```suggestion
        const iconURL = 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png';
```

cwarnermm commented on e2e/cypress/integration/integrations/incoming_webhook/basic_formatting_spec.js:
```diff
@@ -123,6 +123,6 @@ The following should turn into a user mention and clicking it should open profil
         username: 'new_username',
         text,
-        icon_url: 'http://www.mattermost.org/wp-content/uploads/2016/04/icon_WS.png',
+        icon_url: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
        icon_url: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/cypress/integration/integrations/incoming_webhook/inapp_username_profile_override_spec.js:
```diff
@@ -125,7 +125,7 @@ function getPayload(channel, withUsernameAndProfileIcon) {
         ...payload,
         username: 'payload_username',
-        icon_url: 'http://www.mattermost.org/wp-content/uploads/2016/04/icon_WS.png',
+        icon_url: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
        icon_url: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/cypress/integration/integrations/incoming_webhook/slack_formatting_spec.js:
```diff
@@ -134,7 +134,7 @@ describe('Incoming webhook', () => {
             username: 'new_username',
             text,
-            icon_url: 'http://www.mattermost.org/wp-content/uploads/2016/04/icon_WS.png',
+            icon_url: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
            icon_url: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/cypress/integration/integrations/slash_commands_spec.js:
```diff
@@ -195,7 +195,7 @@ describe('Integrations', () => {
 
         // # Post message
-        cy.postMessage('http://mattermost.org/wp-content/uploads/2016/04/icon_WS.png');
+        cy.postMessage('http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png');
```
```suggestion
        cy.postMessage('https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png');
```

cwarnermm commented on e2e/cypress/integration/integrations/slash_commands_spec.js:
```diff
@@ -215,7 +215,7 @@ describe('Integrations', () => {
 
         // # Post message
-        cy.postMessage('http://www.mattermost.org/wp-content/uploads/2016/04/icon_WS.png');
+        cy.postMessage('http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png');
```
```suggestion
        cy.postMessage('https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png');
```

cwarnermm commented on e2e/cypress/integration/markdown/markdown_image_spec.js:
```diff
@@ -145,7 +145,7 @@ describe('Markdown', () => {
     it('channel header is markdown image that is also a link', () => {
         // # Update channel header
-        cy.updateChannelHeader('[![Build Status](https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)](https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)').wait(TIMEOUTS.TWO_SEC);
+        cy.updateChannelHeader('[![Build Status](http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)](http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)').wait(TIMEOUTS.TWO_SEC);
```
```suggestion
        cy.updateChannelHeader('[![Build Status](https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)](https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)').wait(TIMEOUTS.TWO_SEC);
```

cwarnermm commented on e2e/cypress/integration/messaging/inline_images_open_preview_window_spec.js:
```diff
@@ -31,7 +31,7 @@ describe('Messaging', () => {
 
         // # Post the image link to the channel
-        cy.postMessage('Hello ![test image](https://mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)');
+        cy.postMessage('Hello ![test image](http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)');
```
```suggestion
        cy.postMessage('Hello ![test image](https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)');
```

cwarnermm commented on e2e/cypress/integration/notifications/desktop_notifications_1_spec.js:
```diff
@@ -96,7 +96,7 @@ describe('Desktop notifications', () => {
         spyNotificationAs('withNotification', 'granted');
 
-        const actualMsg = '*I\'m* [hungry](http://example.com) :taco: ![Mattermost](http://www.mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)';
+        const actualMsg = '*I\'m* [hungry](http://example.com) :taco: ![Mattermost](http://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)';
```
```suggestion
        const actualMsg = '*I\'m* [hungry](http://example.com) :taco: ![Mattermost](https://mattermost.com/wp-content/uploads/2022/02/logoHorizontal.png)';
```

cwarnermm commented on e2e/utils/report.js:
```diff
@@ -152,11 +152,11 @@ function generateTestReport(summary, isUploadedToS3, reportLink, environment, te
         return {
             username: 'Cypress UI Test',
-            icon_url: 'https://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+            icon_url: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
            icon_url: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/report.js:
```diff
@@ -152,11 +152,11 @@ function generateTestReport(summary, isUploadedToS3, reportLink, environment, te
                 color: testResult.color,
                 author_name: 'Webapp End-to-end Testing',
-                author_icon: 'https://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+                author_icon: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
                author_icon: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/report.js:
```diff
@@ -189,11 +189,11 @@ function generateTestReport(summary, isUploadedToS3, reportLink, environment, te
     return {
         username: 'Cypress UI Test',
-        icon_url: 'https://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+        icon_url: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
        icon_url: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/report.js:
```diff
@@ -254,11 +254,11 @@ function generateDiagnosticReport(summary, serverInfo) {
     return {
         username: 'Cypress UI Test',
-        icon_url: 'https://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+        icon_url: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
        icon_url: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/report.js:
```diff
@@ -254,11 +254,11 @@ function generateDiagnosticReport(summary, serverInfo) {
             color: '#43A047',
             author_name: 'Cypress UI Test',
-            author_icon: 'https://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+            author_icon: 'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
            author_icon: 'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/webhook_utils.js:
```diff
@@ -9,7 +9,7 @@ function getFullDialog(triggerId, webhookBaseUrl) {
             title: 'Title for Full Dialog Test',
             icon_url:
-                'http://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+                'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
                'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/webhook_utils.js:
```diff
@@ -181,7 +181,7 @@ function getSimpleDialog(triggerId, webhookBaseUrl) {
             title: 'Title for Dialog Test without elements',
             icon_url:
-                'http://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+                'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
                'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/webhook_utils.js:
```diff
@@ -197,7 +197,7 @@ function getUserAndChannelDialog(triggerId, webhookBaseUrl) {
             title: 'Title for Dialog Test with user and channel element',
             icon_url:
-                'http://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+                'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
                'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented on e2e/utils/webhook_utils.js:
```diff
@@ -243,7 +243,7 @@ function getBooleanDialog(triggerId, webhookBaseUrl) {
             title: 'Title for Dialog Test with boolean element',
             icon_url:
-                'http://www.mattermost.org/wp-content/uploads/2016/04/icon.png',
+                'http://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```
```suggestion
                'https://mattermost.com/wp-content/uploads/2022/02/icon_WS.png',
```

cwarnermm commented:
@hmhealey and @saturninoabril  - I believe that all paused/outstanding work on this PR has now been completed.

I have removed all of the language files I touched with URL updates (except the ``en.json`` file), and I replaced the ``about.mm.com`` legacy URLs you flagged as potentially problematic with the new destination links Emily made available. Removing the "Do not merge" label for final reviews.

cwarnermm commented:
/update-branch

cwarnermm commented:
@hmhealey - I've un-deleted the language files, as we discussed via DM, by reverting them to last known state in master.

saturninoabril commented:
Thanks @cwarnermm for updating! Will take a look and run E2E once server and enterprise changes were merged.

saturninoabril commented:
Server changes merged. Waiting for enterprise to be merged.

saturninoabril commented:
/e2e-test

mattermod commented:
Triggering e2e testing with options:
```<nil>```

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/171792

saturninoabril commented:
/e2e-test

mattermod commented:
Triggering e2e testing with options:
```<nil>```

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/171828

saturninoabril reviewed (approved):
Thanks @cwarnermm! LGTM. Other failed tests in E2E are not related to this PR.
